bson/codec_options.py

```diff
@@ -32,10 +33,41 @@ def _raw_document_class(document_class):
     return marker == _RAW_BSON_DOCUMENT_MARKER
 
 
+class TypeRegistry(object):
+    def __init__(self, encoder_map=None, decoder_map=None):
+        self.__encoder_map = encoder_map or {}
+        self.__decoder_map = decoder_map or {}
+
+    def copy(self):
+        return type(self)(self.__encoder_map.copy(), self.__decoder_map.copy())
+
+    def _arguments_repr(self):
+        return ('Encode=%r, Decode=%r' % (list(self.__encoder_map.keys()), list(self.__decoder_map.keys())))
+
+    def __repr__(self):
+        return '%s(%s)' % (
+            self.__class__.__name__,
+            'encodable_types=%r, decodable_types=%r' % (
+                [t.__name__ for t in self.__encoder_map.keys()],
+                [t.__name__ for t in self.__decoder_map.keys()]))
```

Comment:
I think it would be clearer to follow the "`repr` returns an executable representation pattern" so the this should use `__init__`'s argument names (`encoder_map/decoder_map`).